itests/org.openhab.binding.modbus.tests/src/main/java/org/openhab/binding/modbus/tests/ModbusDataHandlerTest.java

```diff
@@ -798,8 +797,16 @@ public void testRefreshOnData() throws InterruptedException {
         assertThat(dataHandler.getThing().getStatus(), is(equalTo(ThingStatus.ONLINE)));
 
         verify(comms, never()).submitOneTimePoll(eq(request), notNull(), notNull());
-        // Reset initial REFRESH commands to data thing channels from the Core
+        ModbusPollerThingHandler handler = (ModbusPollerThingHandler) poller.getHandler();
+        // Wait for all channels to receive the REFRESH command (initiated by the core)
+        waitForAssert(
+                () -> verify((ModbusPollerThingHandler) poller.getHandler(), times(CHANNEL_TO_ACCEPTED_TYPE.size()))
+                        .refresh(),
+                2500, 50);
```

Comment:
Is there any reason not relying on the default timeout of `waitForAsser()`?

Reply:
Default should work just as fine I can imagine...